JavaScript/ds/recursive_preorder.js

```diff
@@ -0,0 +1,102 @@
+/*
+In Preorder traversal , the root is visited first , then the let sub-tree in preorder
+fashion, and then the right subtree in preorder fashion.The traversal is defined as 
+* Visit the root node R
+*Traverse the left sub-tree of R in preorder
+*Traverse the right sub-tree of R in preorder
+*/
+
+class Node {
+    constructor(data) {
+        this.data = data;
+        this.left = null;
+        this.right = null;
+    }
+}
+
+class BinarySearchTree {
+    constructor() {
+        this.root = null;
+    }
+    //insert function inserts a new node into the binary search tree
+    insert(value) {
+        var New = new Node(value);
+        if (this.root === null) {
+            this.root = New;
+            return this;
+        }
+        let curr = this.root;
+        let prev = null;
+        while (curr) {
+            if (value < curr.data) {
+                prev = curr;
+                curr = curr.left;
+            }
+            else if (value > curr.data) {
+                prev = curr;
+                curr = curr.right;
+            }
+        }
+        if (prev.data > value) {
+            prev.left = New;
+            return this;
+        }
+        else {
+            prev.right = New;
+            return this;
+        }
+
+    }
+
+    recursive_preorder(node) {
+        if (node !== null) {
+            console.log(node.data);
+            this.recursive_preorder(node.left);
+            this.recursive_preorder(node.right);
+        }
+    }
+}
+
+
+const readline = require("readline-sync");
```

Comment:
You are using a npm library for reading `stdin` if there is a way to do it without a npm package do it that way. Because if you start using npm packages then we will need to maintain a package.json as well with the script which you have written

Reply:
Sorry , I'l use built-in modules.

Reply:
No need to be sorry. Let us know when you fix this

Reply:
@gat786  I have used a build in module for reading input , can you please review my pr